### PR TITLE
gh actions: add workflow for e2e tests

### PR DESCRIPTION
**Pull request by fromanirh**
In this PR we set up the infra to run e2e tests using gh actions, and we backport the e2e tests from https://github.com/k8stopologyawareschedwg/node-feature-discovery/

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      run: |
+        RUNTIME=docker IMAGETAG=ci make image
+
+    - name: generate manifests
+      run: |
+        RTE_CONTAINER_IMAGE=${RTE_CONTAINER_IMAGE} make gen-manifests > rte-e2e.yaml
+
+    - name: create K8S kind cluster
+      run: |
+        kind create cluster --config=hack/kind-config-e2e.yaml --image kindest/node:v1.21.1@sha256:69860bda5563ac81e3c0057d654b5253219618a22ec3a346306239bba8cfa1a6
```
Comment (Tal-or) [inline .github/workflows/e2e.yml:43]: Does ```kind``` pre-installed on this ```ubuntu-20.04``` VM?

File: hack/get-manifest-ds.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+DIRNAME="$(dirname "$(readlink -f "$0")")"
+DEFAULT_IMAGE="quay.io/k8stopologyawarewg/resource-topology-exporter:latest"
+REPOOWNER=${REPOOWNER:-k8stopologyawarewg}
+IMAGENAME=${IMAGENAME:-resource-topology-exporter}
+IMAGETAG=${IMAGETAG:-latest}
+export RTE_CONTAINER_IMAGE=${RTE_CONTAINER_IMAGE:-quay.io/${REPOOWNER}/${IMAGENAME}:${IMAGETAG}}
+envsubst < ${DIRNAME}/../manifests/resource-topology-exporter-ds.yaml
```
Comment (Tal-or) [inline hack/get-manifest-ds.sh:9]: What env variable you're trying to substitute here?

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (fromanirh) [reply .github/workflows/e2e.yml:43]: Yes: https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md

File: hack/get-manifest-ds.sh (hunk 2 above)
Comment (fromanirh) [reply hack/get-manifest-ds.sh:9]: it's `RTE_CONTAINER_IMAGE`: https://github.com/k8stopologyawareschedwg/resource-topology-exporter/blob/master/manifests/resource-topology-exporter-ds.yaml#L45

**Pull request merged**